static/js/components/CentroidPlot.jsx

```diff
@@ -38,351 +55,480 @@ const gcirc = (ra1, dec1, ra2, dec2) => {
-      },
-      mark: {
-        type: "point",
-        shape: "cross",
-        size: "100",
-      },
-      encoding: {
-        x: {
-          field: "delRA",
-          type: "quantitative",
-        },
-        y: {
-          field: "delDec",
-          type: "quantitative",
-        },
-        fill: { value: "black" },
-      },
-    },
-
-    // Render text messages
-    {
-      data: {
-        values: inputData.messages,
-      },
-      mark: {
-        type: "text",
-        fontSize: 14,
-        fontWeight: 500,
-      },
-      encoding: {
-        text: { field: "message", type: "nominal" },
-        color: { value: textColor },
-        x: {
-          field: "x",
-          type: "quantitative",
-        },
-        y: {
-          field: "y",
-          type: "quantitative",
-        },
-      },
-    },
-  ],
-});
-
-const processData = (photometry) => {
-  // Only take points with a non-null RA and Dec
-  const filteredPhotometry = photometry?.filter(
-    (point) => point.ra && point.dec,
+const prepareData = (photometry, fallbackRA, fallbackDec) => {
+  console.log(photometry);
+  if (!photometry || photometry.length === 0) {
+    return { refRA: null, refDec: null, oneSigmaCircle: null, stdCircle: null };
+  }
+  // keep only the points with a mag and magerr
+  // and ra, dec
+  // and that are not forced photometry
+  let points = photometry.filter(
+    (p) =>
+      p.mag !== null &&
+      p.magerr !== null &&
+      p.ra !== null &&
+      p.dec !== null &&
+      !Number.isNaN(p.mag) &&
+      !Number.isNaN(p.magerr) &&
+      !Number.isNaN(p.ra) &&
+      !Number.isNaN(p.dec) &&
+      !["fp", "alert_fp"].includes(p.origin),
   );
-
-  if (filteredPhotometry.length === 0) {
+  if (points.length === 0) {
     return {
-      photometryData: [],
+      refRA: null,
+      refDec: null,
+      points: [],
+      oneSigmaCircle: null,
+      stdCircle: null,
     };
   }
-
-  const ras = Object.values(filteredPhotometry).map((point) => point.ra);
-  const decs = Object.values(filteredPhotometry).map((point) => point.dec);
-
-  // For now, set single reference nearest object to median values for the RA
-  // and Dec in the photometry
-  const { refRA, refDec } = getReferencePoint(ras, decs);
-
-  const computeDeltas = (delRaGroup, delDecGroup) => (point) => {
-    const { delRA, delDec } = relativeCoord(point.ra, point.dec, refRA, refDec);
-    delRaGroup.push(delRA);
-    delDecGroup.push(delDec);
+  // calculate the flux and fluxerr for these points
+  points = points.map((p) => {
+    const newPoint = { ...p };
+    newPoint.flux = 10 ** (-0.4 * (p.mag - PHOT_ZP));
+    newPoint.fluxerr = (newPoint.magerr / (2.5 / Math.log(10))) * newPoint.flux;
+    newPoint.snr = newPoint.flux / newPoint.fluxerr;
+    return newPoint;
+  });
+  // remove those with a snr < 3.0 and valid flux and fluxerr
+  points = points.filter(
+    (p) => p.snr >= 3.0 && p.flux !== null && p.fluxerr !== null,
```

Comment:
Maybe make 3.0 a variable?

Reply:
smort. Let me do that!

Reply:
It is fine if I make it a constant at the top of the file for now, and add a slider/number input in a future PR? I would love to get this one in today.

Reply:
Of
Course